web-client/src/primitives/transaction_builder.rs

```diff
@@ -251,7 +256,43 @@ impl TransactionBuilder {
         Ok(Transaction::from_native(tx))
     }
 
-    // pub fn new_create_validator()
+    /// Registers a new validator in the staking contract.
+    ///
+    /// The returned transaction is not yet signed. You can sign it e.g. with `tx.sign(keyPair)`.
+    ///
+    /// Throws when the numbers given for fee do not fit within a u64 or the networkId is unknown.
+
+    #[wasm_bindgen(js_name = newCreateValidator)]
+    pub fn new_create_validator(
+        sender: &Address,
+        reward_address: &Address,
+        signing_key: PublicKey,
+        voting_key_pair: BLSKeyPair,
+        fee: Option<u64>,
+        validity_start_height: u32,
+        network_id: u8,
+    ) -> Result<Transaction, JsError> {
+        let mut recipient = Recipient::new_staking_builder();
+        recipient.create_validator(
+            signing_key.native_ref().clone(),
+            &voting_key_pair.native_ref().clone(),
+            reward_address.native_ref().clone(),
+            None,
```

Comment:
Is there anything speaking against giving the option to set signal data?

Reply:
This would have required to implement OptionFromWasmAbi for the Blake2bHash. As this option was not needed for me at the time I skipped it.

Reply:
More transactions are work in progress:
https://github.com/nimiq/core-rs-albatross/pull/1870

Reply:
Signal data is not a hash, is it? It's any bytes. So either `Option<&[u8]>` or `Option<String>` for hex would have worked.